File: backend/app/schema/project.py
from datetime import datetime
from typing import Optional
from uuid import UUID

from pydantic import BaseModel


class ProjectCreate(BaseModel):
    name: str
    description: Optional[str] = None


class ProjectUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None


class ProjectResponse(BaseModel):
    id: UUID
    name: str
    description: Optional[str] = None
    owner_id: UUID
    created_at: datetime
    updated_at: datetime

File: frontend/src/types/project.ts
export interface ProjectCreate {
	name: string;
	description?: string | null;
}

export interface ProjectUpdate {
	name?: string | null;
	description?: string | null;
}

export interface ProjectResponse {
	id: string;
	name: string;
	description?: string | null;
	owner_id: string;
	created_at: string;
	updated_at: string;
}
